internal/common/objects/BaseBiker.go

```diff
@@ -136,32 +136,72 @@ func (bb *BaseBiker) ResetLootAttributes() {
 	bb.resourceAppropriation = 1
 }
 
-//
+// the biker itself doesn't technically have a location (as it's on the map only when it's on a bike)
+// in fact this function is only called when the biker needs to make a decision about the pedaling forces
+func (bb *BaseBiker) GetLocation() utils.Coordinates {
+	megaBikes := bb.gameState.GetMegaBikes()
+	return megaBikes[bb.megaBikeId].coordinates
+}
+
+// returns the nearest lootbox with respect to the agent's bike current position
+// in the MVP this is used to determine the pedalling forces as all agent will be
+// aiming to get to the closest lootbox by default
+func (bb *BaseBiker) NearestLoot() utils.Coordinates {
```

Comment:
Perhaps after the MVP, we can return the lootbox objects in ranked order of nearest, so every new type of agent will not have to implement their own method for this.